File: day-02/c/part1.c
#include <stdio.h>
#include <stdbool.h>
#include "../../utils/io_utils.h"

#define M 100

int main(void) {
    int reports[M], n, order, i, diff, safe_count = 0;
    bool safe;
    while ((n = read_ints(stdin, reports))) {
        safe = true;
        if (n > 1) {
            order = reports[1] > reports[0] ? 1 : -1;
            for (i = 1; i < n; i++) {
                diff = reports[i] - reports[i - 1];
                if (order == 1) {
                    if (!(diff >= 1 && diff <= 3)) {
                        safe = false;
                        break;
                    }
                } else {
                    if (!(diff >= -3 && diff <= -1)) {
                        safe = false;
                        break;
                    }
                }
            }
            if (safe) safe_count++;
        }
    }
    printf("%d\n", safe_count);
    return 0;
}

File: day-02/py/part1.py
with open("input.txt", "r") as file:
    safe_count = 0
    for line in file:
        reports = [int(x) for x in line.strip().split()]
        safe = True
        if len(reports) > 1:
            order = 1 if reports[1] > reports[0] else -1
            for i in range(1, len(reports)):
                diff = reports[i] - reports[i - 1]
                if order == 1:
                    if not (1 <= diff <= 3):
                        safe = False
                        break
                else:
                    if not (-3 <= diff <= -1):
                        safe = False
                        break
        if safe:
            safe_count += 1

print(safe_count)